ui/src/criteria/attribute.tsx

```diff
@@ -410,8 +410,8 @@ function AttributeDetails(props: AttributeDetailsProps) {
   if (props.data.selected?.length) {
     return (
       <>
-        {props.data.selected.map(({ id, name }) => (
-          <Stack direction="row" alignItems="baseline" key={Date.now()}>
+        {props.data.selected.map(({ id, name }, index) => (
+          <Stack direction="row" alignItems="baseline" key={index}>
```

Comment:
This shouldn't be in this PR. Also, this is incorrect for changes other than simple adds or deletes at the end of the list. Any reordering or editing elsewhere in the list will result in keys mapping to different items.